transport/http/outbound.go

```diff
@@ -66,6 +66,12 @@ func NewOutbound(url string, opts ...OutboundOption) transport.UnaryOutbound {
 	return newOutbound(url, opts...).(transport.UnaryOutbound)
 }
 
+// NewOnewayOutbound builds a new HTTP outbound that sends oneway requests to
+// the given URL.
+func NewOnewayOutbound(url string, opts ...OutboundOption) transport.OnewayOutbound {
+	return newOutbound(url, opts...).(transport.OnewayOutbound)
```

Comment:
Instead of casting, change the internal `newOutbound` method to return a
`*outbound`. Since the type implements both interfaces, you can simply return
it as-is from both functions.

Also, we may want to just expose the type rather than two functions with the
same body.

```
package http

type Outbound struct {
    // all fields private
}

// This just verifies that the type implements both interfaces
var (
    _ transport.UnaryOutbound = (*Outbound)(nil)
    _ transport.OnewayOutbound = (*Outbound)(nil)
)

func NewOutbound(...) *Outbound {
    // ..
}
```

Callers can then just use that object in both places:

```
h := http.NewOutbound(...)
yarpc.Config{
    Outbounds: yarpc.Outbounds{
      "foo": {Unary: h, Oneway: h},
    },
}
```